src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -462,6 +485,8 @@ public void getDocIds(DocIdPusher pusher) throws IOException,
       log.finer("queried for stream");
       boolean hasAction
           = hasColumn(rs.getMetaData(), GsaSpecialColumns.GSA_ACTION);
+      boolean hasTimestamp =
```

Comment:
I thought we were going to clear the cache in getDocIds (but not getModifiedDocIds, obviously). That avoids keeping a record for deleted rows, and in particular, if the ordering in getDocContent is fixed, avoids us reporting deleted rows as unmodified instead.